homeassistant/components/camera/__init__.py

```diff
@@ -207,6 +218,12 @@ def state_attributes(self):
 
         return attr
 
+    @callback
+    def async_update_token(self):
+        """Update the used token."""
+        self.access_tokens.append(
+            hashlib.sha256(str.encode(str(_RND.random()))).hexdigest())
```

Comment:
Additionally, the code would be DRYer if there was a generate_auth_token util method. It would make it easier to tweak the token generation mechanism in the future, and would discourage people from rolling their own method for making tokens.